core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregator.java

```diff
@@ -277,32 +280,151 @@ public void doWriteTo(StreamOutput out) throws IOException {
 
     public static class Factory extends PipelineAggregatorFactory {
 
-        private final ValueFormatter formatter;
-        private GapPolicy gapPolicy;
-        private int window;
-        private MovAvgModel model;
-        private int predict;
-        private boolean minimize;
+        private String format;
+        private GapPolicy gapPolicy = GapPolicy.SKIP;
+        private int window = 5;
+        private MovAvgModel model = new SimpleModel();
+        private int predict = 0;
+        private Boolean minimize;
 
-        public Factory(String name, String[] bucketsPaths, ValueFormatter formatter, GapPolicy gapPolicy,
-                       int window, int predict, MovAvgModel model, boolean minimize) {
+        public Factory(String name, String[] bucketsPaths) {
             super(name, TYPE.name(), bucketsPaths);
-            this.formatter = formatter;
+        }
+
+        /**
+         * Sets the format to use on the output of this aggregation.
+         */
+        public void format(String format) {
+            this.format = format;
+        }
+
+        /**
+         * Gets the format to use on the output of this aggregation.
+         */
+        public String format() {
+            return format;
+        }
+
+        /**
+         * Sets the GapPolicy to use on the output of this aggregation.
+         */
+        public void gapPolicy(GapPolicy gapPolicy) {
             this.gapPolicy = gapPolicy;
+        }
+
+        /**
+         * Gets the GapPolicy to use on the output of this aggregation.
+         */
+        public GapPolicy gapPolicy() {
+            return gapPolicy;
+        }
+
+        protected ValueFormatter formatter() {
+            if (format != null) {
+                return ValueFormat.Patternable.Number.format(format).formatter();
```

Comment:
I don't know how often this is called, depending on this maybe it makes sense to store the formatter somewhere for later reuse unless `format` changes? Is only called a few times maybe not worth the trouble.